### PR TITLE
Updated kpoints.in and basis.in

### PR DESCRIPTION
**Pull request by snadee001**
### Purpose:
To create the kpoints.in (generates k-points needed) and the basis.in (defines the unit cell for phonon calculations and maps atoms in supercell to basis atoms in unit cell) files. 

### Details:
Create_kpoints requires the user to take in an ase.Atoms object, special points, and an int value (for the number of k points to generate). Using the ASE function, cell.bandpath, an array of k-points is created. The create_kpoints function cleans up the array and places the information in a file named kpoints.in

Create_basis takes in an atoms object (the unit cell) and a transformation vector (a, b, c). From this, it creates a supercell from the unit cell and maps all the atoms in the supercell to its matching atom in the unit cell. It then generates the basis.in file, which contains the number of atoms in the unit cell (as well as their index and mass) and then lists the index of all other atoms in the supercell.

### CODE REVIEW

File: thermo/gpumd/io.py [hunk 1]
```diff
@@ -249,6 +249,61 @@ def import_trajectory(filename='movie.xyz', in_file=None, atom_types=None):
 # Write Related
 #########################################
 
+def create_kpoints(special_points, npoints, atom_c):
```
Comment (AlexGabourie) [inline thermo/gpumd/io.py]:
```suggestion
def create_kpoints(atoms, special_points = 'G', npoints=1):
```

File: thermo/gpumd/io.py [hunk 2]
```diff
@@ -249,6 +249,61 @@ def import_trajectory(filename='movie.xyz', in_file=None, atom_types=None):
...
+    """
+     Creates the file "kpoints.in", which specifies the kpoints needed for src/phonon
+
+    Args:
+        atom_c (ase.Atom):
+            atom to build
```
Comment (AlexGabourie) [inline thermo/gpumd/io.py]:
```suggestion
        atoms (ase.Atoms):
            Atoms used to generate kpoints.in.
```

File: thermo/gpumd/io.py (hunk 1 above)
Comment (AlexGabourie) [reply thermo/gpumd/io.py]: Order of the input arguments should match that in the comments below. Usually in ASE, the first parameter is the Atoms object. I have also added some default parameters. Check to make sure that these work.

File: thermo/gpumd/io.py (hunk 2 above)
Comment (AlexGabourie) [reply thermo/gpumd/io.py]: The user should be providing an Atoms object, not Atom. I changed the variable name to match the convention used in ASE.

File: thermo/gpumd/io.py [hunk 3]
```diff
@@ -249,6 +249,61 @@ def import_trajectory(filename='movie.xyz', in_file=None, atom_types=None):
...
+
+        special_points (str):
+            specify special points for kpoint generation
+
+        npoints (int):
+            specifies the number of kpoints to generate
+    """
+
+    path = atom_c.cell.bandpath(special_points, npoints)
```
Comment (AlexGabourie) [inline thermo/gpumd/io.py]:
```suggestion
    path = atoms.cell.bandpath(special_points, npoints)
```
Updated variable name to match above

File: thermo/gpumd/io.py [hunk 4]
```diff
@@ -249,6 +249,61 @@ def import_trajectory(filename='movie.xyz', in_file=None, atom_types=None):
...
+    kpts = str(path.kpts)
+    kpts = kpts.replace("[[", "")
+    kpts = kpts.replace(" [", "")
+    kpts = kpts.replace("]", "")
+    kpts = kpts.replace("0. ", "0  ")
+
+    with open("kpoints.in", "w") as f:
+        f.write(s_str + "\n" + kpts)
+
+    return
```
Comment (AlexGabourie) [inline thermo/gpumd/io.py]: return statement not needed if not returning anything

File: thermo/gpumd/io.py [hunk 5]
```diff
@@ -249,6 +249,61 @@ def import_trajectory(filename='movie.xyz', in_file=None, atom_types=None):
...
+
+def create_basis(atom, transform):
```
Comment (AlexGabourie) [inline thermo/gpumd/io.py]:
```suggestion
def create_basis(atoms, transform):
```
Using Atoms not Atom

File: thermo/gpumd/io.py [hunk 6]
```diff
@@ -249,6 +249,61 @@ def import_trajectory(filename='movie.xyz', in_file=None, atom_types=None):
...
+    """
+    Creates the file "basis.in", which maps atoms in the super cell to the atoms in the unit cell
```
Comment (AlexGabourie) [inline thermo/gpumd/io.py]:
```suggestion
    Creates the file "basis.in", which maps atoms in the supercell to the atoms in the unit cell
```

File: thermo/gpumd/io.py [hunk 7]
```diff
@@ -249,6 +249,61 @@ def import_trajectory(filename='movie.xyz', in_file=None, atom_types=None):
...
+
+    Args:
+        atom (ase.Atom):
+            atom to build
```
Comment (AlexGabourie) [inline thermo/gpumd/io.py]:
```suggestion
        atoms (ase.Atoms):
            Atoms of unit cell used to generate basis.in
```

File: thermo/gpumd/io.py [hunk 8]
```diff
@@ -249,6 +249,61 @@ def import_trajectory(filename='movie.xyz', in_file=None, atom_types=None):
...
+
+        transform (int | 3 floats):
+            creates the super cell by making the unit cell larger by these 3 amounts.
```
Comment (AlexGabourie) [inline thermo/gpumd/io.py]:
```suggestion
            Creates the supercell by repeating the unit cell by these 3 amounts.
```

**Review by AlexGabourie — CHANGES_REQUESTED** [review]
The code looks good. I suggested some changes that are purely aesthetic. Please make sure you are looking at the comments in the rest of my code to make sure the style is consistent when they are formatted into the [html page](https://thermomd.readthedocs.io/en/latest/index.html). Also, for function parameter names, look to the ASE documentation and try your best to stick to their conventions. For example, if a function takes and Atoms object, the first parameter is usually called 'atoms'. 

Once you make these changes, I will accept the pull request.

**Comment by snadee001** [comment]
I have fixed your comments and a few other stylistic issues with the code!

**Review by AlexGabourie — APPROVED** [review]

**Pull request merged**